lib/matplotlib/axis.py

```diff
@@ -251,7 +251,9 @@ def set_pad(self, val):
         """
         Set the tick label pad in points
 
-        ACCEPTS: float
+        Parameters
+        ----------
+        val : float
```

Comment:
Would it make sense to use type annotations for these very simple cases? The type could be inferred from the type annotation. We could leave out the Parameters section which is quite verbose just for specifying a simple type, both in source code and in the generated html.

Reply:
That's a discussion for another time...

Reply:
Well, just saying. It's just one line in the `ArtistInspector` (actually none, because you can delete another line for that). And it saves a lot of boilerplate docstring code.

Reply:
I'd be more comfortable if we can draw a policy as to where to draw the line between strict annotations and "vague" type docstrings.  Especially for Matplotlib, there are many cases where the exact type is complicated or it's actually the value (rather than just the type) that's important.

Reply:
The policy for the docstring is: Describes the expected input parameters in a way, that is easy to read and understand by a human user. The numpydoc howto gives some helpful guidelines that should be followed if possibe.

I currently would only use type annotations for simple cases as the one above. Anyway, should we have both later on, the docstring type should be given preference. It's really only that you don't need verbose boilerplate code like

~~~
Parameters
------------
theparam : float
~~~

But I don't hold you up on this.